src/main/java/io/github/pylonmc/pylon/base/items/Grindstone.java

```diff
@@ -0,0 +1,336 @@
+
+            if (recipe != null) {
+                return;
+            }
+
+            ItemDisplay itemDisplay = getItemDisplay();
+            ItemStack oldStack = itemDisplay.getItemStack();
+            ItemStack newStack = event.getItem();
+
+            // drop old item
+            if (!oldStack.getType().isAir()) {
+                getBlock().getWorld().dropItem(getBlock().getLocation().toCenterLocation().add(0, 0.25, 0), oldStack);
+                itemDisplay.setItemStack(null);
+                return;
+            }
+
+            // insert new item
+            if (newStack != null) {
+                ItemStack stackToInsert = newStack.clone();
+                itemDisplay.setItemStack(stackToInsert);
+                newStack.setAmount(0);
+            }
+        }
+
+        @Override
+        public void onBreak(@NotNull List<ItemStack> drops, @NotNull BlockBreakContext context) {
+            SimplePylonMultiblock.super.onBreak(drops, context);
+            drops.add(getItemDisplay().getItemStack());
+        }
+
+        @Override
+        public int getCustomTickRate(int globalTickRate) {
+            return TICK_RATE;
+        }
+
+        @Override
+        public void tick(double deltaSeconds) {
+            if (!isFormedAndFullyLoaded() || cyclesRemaining == null || recipe == null || cycleTicksRemaining == null) {
+                return;
+            }
+
+            // check if cycle finished
+            if (cycleTicksRemaining <= 0) {
+                cycleTicksRemaining = CYCLE_TIME_TICKS;
+                cyclesRemaining -= 1;
+
+                // check if recipe finished
+                if (cyclesRemaining <= 0) {
+
+                    // rotate stone back to starting rotation
+                    getStoneDisplay().setTransformationMatrix(new TransformBuilder()
+                            .translate(0, 0.5, 0)
+                            .buildForItemDisplay());
+                    getStoneDisplay().setInterpolationDelay(0);
+                    getStoneDisplay().setInterpolationDuration(TICK_RATE);
+
+                    cyclesRemaining = null;
+                    cycleTicksRemaining = null;
+
+                    // run after interpolation to starting rotation is finished
+                    Bukkit.getScheduler().runTaskLater(PylonBase.getInstance(), this::finishRecipe, TICK_RATE);
+
+                    return;
+                }
+            }
+
+            if (cycleTicksRemaining == CYCLE_TIME_TICKS) {
+                // put top stone down
+                getStoneDisplay().setTransformationMatrix(new TransformBuilder()
+                        .translate(0, 0.5, 0)
+                        .buildForItemDisplay());
+            } else {
+                // rotate stone
+                getStoneDisplay().setTransformationMatrix(new TransformBuilder()
+                        .translate(0, 0.5, 0)
+                        .rotate(0, 2 * Math.PI * ((double) cycleTicksRemaining / CYCLE_TIME_TICKS), 0)
```

Comment:
```suggestion
                        .rotate(0, Math.TAU * ((double) cycleTicksRemaining / CYCLE_TIME_TICKS), 0)
```
I'm only being half serious, feel free to ignore :P